src/writer.rs

```diff
@@ -287,7 +287,11 @@ impl FdtWriter {
 
     /// Write a property containing an array of 32-bit unsigned integers.
     pub fn property_array_u32(&mut self, name: &str, cells: &[u32]) -> Result<()> {
-        let mut arr = Vec::with_capacity(cells.len() * size_of::<u32>());
+        let size = cells
+            .len()
+            .checked_mul(size_of::<u32>())
```

Comment:
I wouldn't object to including this check, but it seems like it should not be possible for it to fail, since `cells` already exists with the corresponding size (maybe it would be clearer to use `std::mem::size_of_val`?)

Reply:
I used `std::mem::size_of_val` as you suggested.